main.js

```diff
@@ -107,25 +108,35 @@ function findIdeaIndex(e) {
 
 function handleFocusOut(e) {
   if (e.target.className === 'card-title' || e.target.className === 'card-body') {
-    console.log("event:", event);
     var newTitle = e.target.closest('.card-content').querySelector('.card-title').innerText;
     var newBody = e.target.closest('.card-content').querySelector('.card-body').innerText;
     var index = findIdeaIndex(e);
 
     ideasArray[index].updateIdea(newTitle, newBody);
     ideasArray[index].saveToStorage(ideasArray);
-    console.log("index:", index);
     // console.log(title.innerText);
     // console.log(body.innerText);
   }
 }
 
 function toggleStar(e) {
-  if (e.target.id === 'btn-star') {
-    var on = document.querySelector('.star-active').classList.remove('hidden');
-    var off = document.querySelector('.star').classList.add('hidden');
+  if(event.target.classList.contains('img-star')) {
+    var getIdea = ideasArray[findIdeaIndex(e)];
+    getIdea.updateStar();
+    var changeStar = getIdea.star ? 'images/star-active.svg' : 'images/star.svg';
+    event.target.setAttribute('src', changeStar);
```

Comment:
We need to use .closest to target the nearest idea-card class so that we can click on all of the cards stars. Something like this or you can just edit what you have above!

`e.target.closest('.idea-card').querySelector('.img-star').src = changeStar;`